### PR TITLE
Take FB quick reply payload rather than text as input

### LINKED ISSUE #3590: Facebook Quick replies doens't work
**Rasa version**: 1.0.0

**Python version**: 3.7.0

**Operating system** (windows, osx, ...):Ubuntu Linux 

**Issue**: I get the following error When adding Quick replies for messenger : 
```
Traceback (most recent call last):
  File "/home/rogers/paysikabot/lib/python3.7/site-packages/rasa/core/channels/facebook.py", line 94, in _handle_user_message
    await self.on_new_message(user_msg)
  File "/home/rogers/paysikabot/lib/python3.7/site-packages/rasa/core/channels/channel.py", line 68, in handler
    await app.agent.handle_message(*args, **kwargs)
  File "/home/rogers/paysikabot/lib/python3.7/site-packages/rasa/core/agent.py", line 441, in handle_message
    return await processor.handle_message(message)
  File "/home/rogers/paysikabot/lib/python3.7/site-packages/rasa/core/processor.py", line 74, in handle_message
    await self._predict_and_execute_next_action(message, tracker)
  File "/home/rogers/paysikabot/lib/python3.7/site-packages/rasa/core/processor.py", line 342, in _predict_and_execute_next_action
    action, tracker, message.output_channel, self.nlg, policy, confidence
  File "/home/rogers/paysikabot/lib/python3.7/site-packages/rasa/core/processor.py", line 453, in _run_action
    await self._send_bot_messages(events, tracker, output_channel)
  File "/home/rogers/paysikabot/lib/python3.7/site-packages/rasa/core/processor.py", line 374, in _send_bot_messages
    await output_channel.send_response(tracker.sender_id, e.message())
  File "/home/rogers/paysikabot/lib/python3.7/site-packages/rasa/core/channels/channel.py", line 173, in send_response
    **message
  File "/home/rogers/paysikabot/lib/python3.7/site-packages/rasa/core/channels/facebook.py", line 181, in send_quick_replies
    self.send(recipient_id, FBText(text=text, quick_replies=quick_replies))
  File "/home/rogers/paysikabot/lib/python3.7/site-packages/rasa/core/channels/facebook.py", line 120, in send
    self.messenger_client.send(element.to_dict(), recipient_id, "RESPONSE")
  File "/home/rogers/paysikabot/lib/python3.7/site-packages/fbmessenger/elements.py", line 27, in to_dict
    d['quick_replies'] = self.quick_replies.to_dict()
AttributeError: 'list' object has no attribute 'to_dict'
```

The error is due to the fact that you are using elements module from _**fbmessenger**_ and in this element module, there is a  Text class, which takes as constructor parameter, a _quick_replies_. But this _quick_replies_  must be a list of _QuickReply_ object. So you are not converting the information provide for Quick replies from the domain.yml into QuickReply Object. Kind of.  

**Content of configuration file (config.yml)**:
```yml
language: en
pipeline: supervised_embeddings   #  pretrained_embeddings_spacy

# Configuration for Rasa Core.
# https://rasa.com/docs/rasa/core/policies/
policies:
  - name: MemoizationPolicy
  - name: KerasPolicy
  - name: FormPolicy
```

**Content of domain file (domain.yml)** (if used & relevant):
```yaml

```

### PR DESCRIPTION
**Pull request by lahsuk**
**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)
- [x] signed CLA

### CODE REVIEW

File: rasa/core/channels/facebook.py [hunk 1]
```diff
@@ -51,6 +51,15 @@ def _is_user_message(message: Dict[Text, Any]) -> bool:
             and not message["message"].get("is_echo")
         )
 
+    @staticmethod
+    def _is_quick_reply_message(message: Dict[Text, Any]) -> bool:
+        """Check if the message is a quick reply message."""
+        return (
+            message.get("message")
```
Comment (tmbo) [inline rasa/core/channels/facebook.py]: let's make this a proper boolean expression (type checker likes that more)
```suggestion
            message.get("message") is not None
```

File: rasa/core/channels/facebook.py [hunk 2]
```diff
@@ -51,6 +51,15 @@ def _is_user_message(message: Dict[Text, Any]) -> bool:
...
+            and message["message"].get("quick_reply")
```
Comment (tmbo) [inline rasa/core/channels/facebook.py]:
```suggestion
            and message["message"].get("quick_reply") is not None
```

**Comment by erohmensing** [comment]
Hey @lahsuk, thanks for this fix! This doesn't actually address the issues you said were closed in the issue (that's an output channel problem), so I removed that, but this is also a very useful contribution. 🙂

**Comment by lahsuk** [comment]
I see.
I send quick replies using custom_json so I, wrongly, assumed that it was a problem after the quick reply buttons were clicked.

**Comment by erohmensing** [comment]
@lahsuk sorry, I should have been clearer, what you did, whether you meant to or not, was find a bug that is difficult to find, since it doesn't usually break anything. Currently, the input channel takes the "text" (i.e. the title) of the quick reply button, instead of the payload, as input. Your fix makes it correctly take the payload as input. We would still love to merge your contribution after you add your fix to the changelog!

**Comment by lahsuk** [comment]
Okay.
It's my first time contributing so I was quite confused. Thank you for clarifying. 😅

**Comment by erohmensing** [comment]
Thank you for your first contribution @lahsuk! 🚀Congrats on joining our group of > 300 contributors 🙂

**Comment by lahsuk** [comment]
Thanks for the warm welcome @erohmensing . 🙏
Hope I can make some more contributions in the coming months. ✨

**Pull request merged**